Sources/Rendering/OpenGL/Texture/index.d.ts

```diff
@@ -0,0 +1,316 @@
+import {Wrap, Filter} from "./Constants";
+import vtkOpenGLRenderWindow from '../RenderWindow';
+import {Nullable} from '../../../types';
+import { vtkAlgorithm, vtkObject } from "../../../interfaces";
+import { VtkDataTypes } from "../../../Common/Core/DataArray";
+
+/**
+ * Represents the time with a modification timestamp.
+ */
+interface Time{
+  mtime:number;
+}
+
+/**
+ * Initial values for creating a new instance of vtkOpenGLTexture.
+ */
+export interface ITextureInitialValues {
+  _openGLRenderWindow: Nullable<vtkOpenGLRenderWindow>;
+  _forceInternalFormat: boolean;
+  context?: WebGLRenderingContext | WebGL2RenderingContext;
+  handle: number;
+  sendParametersTime: Time;
+  textureBuildTime: Time;
+  numberOfDimensions: number;
+  target: number;
+  format: number;
+  openGLDataType: number;
+  components: number;
+  width: number;
+  height: number;
+  depth: number;
+  autoParameters: boolean;
+  wrapS: Wrap.CLAMP_TO_EDGE;
+  wrapT: Wrap.CLAMP_TO_EDGE;
+  wrapR: Wrap.CLAMP_TO_EDGE;
+  minificationFilter: Filter.NEAREST;
+  magnificationFilter: Filter.NEAREST;
+  minLOD: number;
+  maxLOD: number;
+  baseLevel: number;
+  maxLevel: number;
+  generateMipmap: boolean;
+  useHalfFloat: boolean;
+  oglNorm16Ext: null | undefined;
+  allocatedGPUMemoryInBytes: number;
+}
+
+/**
+ * Base type for vtkOpenGLTexture excluding the 'set' method from vtkObject.
+ */
+type vtkOpenGLTextureBase = Omit<vtkObject, 'set'> & vtkAlgorithm;
+
+/**
+ * Interface for OpenGL Texture.
+ */
+export interface vtkOpenGLTexture extends vtkOpenGLTextureBase {
```

Comment:
This should be `vtkOpenGLTexture extends vtkViewNode`, based on what vtkOpenGLTexture extends in the codebase.